src/wallet/walletdb.cpp

```diff
@@ -214,36 +214,40 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
         } else if (strType == DBKeys::TX) {
             uint256 hash;
             ssKey >> hash;
-            CWalletTx wtx(nullptr /* pwallet */, MakeTransactionRef());
-            ssValue >> wtx;
-            if (wtx.GetHash() != hash)
-                return false;
+            auto update_wtx = [&](CWalletTx& wtx, bool /* new_tx */) {
```

Comment:
In 754eb2112aac4729805dcc547bef1de956c3c14d `Avoid copying CWalletTx in LoadToWallet`

Maybe add a comment here that `LoadToWallet` creates a fresh `CWalletTx` and that we fill it with `ssValue` (hence my rename to `fill_wtx` suggestion above).

Since a wallet doesn't (shouldn't? otherwise add `assert(new_tx)`) have duplicate transactions `new_tx` should be `true`, but if some reason a wallet entry does exists it will be overridden.